django/db/migrations/operations/models.py

```diff
@@ -929,6 +929,22 @@ def migration_name_fragment(self):
     def reduce(self, operation, app_label):
         if isinstance(operation, RemoveIndex) and self.index.name == operation.name:
             return []
+        if (
+            isinstance(operation, RenameIndex)
+            and operation.old_name
+            and operation.old_name == self.index.name
```

Comment:
```suggestion
            and operation.old_name == self.index.name
```
I believe this is unnecessary as `old_name` would be `None` then `None == <name>` would fail 👍 (Indexes always end up with a name)